feeder/src/vote.ts

```diff
@@ -74,7 +74,46 @@ interface Price {
   price: string
 }
 
-async function getPrices(sources: string[]): Promise<Price[]> {
+function calculateSDR(prices: Price[], sdrBasket: string): Price | undefined {
+  if (!sdrBasket) {
+    return undefined
+  }
+
+  // check if all prices from the basket are available
+  for (const denom of Object.keys(sdrBasket)) {
+    if (!prices.find((p) => p.denom === denom)) {
+      logger.error(`getPrices: price for ${denom} not found`)
+      return undefined
+    }
+  }
+
+  // calculate SDR price
+  let sdrPrice: BigNumber = undefined
+
+  try {
+    sdrPrice = Object.entries(sdrBasket).reduce((acc, [denom, weight]) => {
+      const price = prices.find((p) => p.denom === denom)
+      if (!price) {
+        throw new Error(`price for ${denom} not found`)
+      }
+      return acc.plus(new BigNumber(price.price).times(weight))
+    }, new BigNumber(0))
+  } catch (err) {
+    logger.error(`getPrices: error calculating SDR price: ${err.message}`)
+    return undefined
+  }
+
+  if (!sdrPrice) {
+    return undefined
```

Comment:
A good test case here. 
Also worth testing feeds with multiple provider setup (multiple vals with different providers) just to be on the safe side.